Kerberos.NET/Replay/MemoryTicketCache.cs

```diff
@@ -2,95 +2,65 @@
 using System.Collections.Concurrent;
 using System.Diagnostics;
 using System.Linq;
-using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 
 namespace Kerberos.NET
 {
-    internal sealed class MemoryTicketCache : ITicketCache, IDisposable
+    internal sealed class MemoryTicketCache : TicketCacheBase
     {
-        private readonly ILogger logger;
-        private readonly Task backgroundRunner;
-        private readonly CancellationTokenSource cts;
-
         public MemoryTicketCache(ILoggerFactory logger)
+            : base(logger)
         {
-            this.logger = logger.CreateLoggerSafe<MemoryTicketCache>();
-
-            cts = new CancellationTokenSource();
-
-            backgroundRunner = Task.Run(RunBackground, cts.Token);
         }
 
-        internal Func<CacheEntry, Task> Refresh { get; set; }
-
-        public bool RefreshTickets { get; set; }
-
-        public TimeSpan RefreshInterval { get; set; } = TimeSpan.FromSeconds(30);
-
-        private async Task RunBackground()
-        {
-            while (true)
-            {
-                if (cts.Token.IsCancellationRequested)
-                {
-                    break;
-                }
-
-                try
-                {
-                    await BackgroundCacheOperation();
-                }
-                catch (Exception ex)
-                {
-                    logger.LogWarning(ex, "Background cache operation failed");
-                }
+        private readonly ConcurrentDictionary<string, CacheEntry> cache
```

Comment:
```suggestion
        private readonly ConcurrentDictionary<string, CacheEntry> _cache
```

Reply:
Underscores are against the soon to be added stylecop guide.

Reply:
Ah, thanks for the info.